src/services/completions.ts

```diff
@@ -2197,4 +2198,28 @@ namespace ts.Completions {
     function hasIndexSignature(type: Type): boolean {
         return !!type.getStringIndexType() || !!type.getNumberIndexType();
     }
+
+    function isInvalidTrigger(sourceFile: SourceFile, triggerCharacter: string | undefined, contextToken: Node, position: number): boolean {

```

Comment:
consider switching this to `isValidTrigger`, and changing the condition to be `if (triggerCharacter && !isValid()) return undefined`